### PR TITLE
[DO NOT MERGE] fixup! 00251: Change user install location

### PR DESCRIPTION
**Pull request by frenzymadness**
This is an udpated version of patch 251 for Python 3.10 + 3.11 in rawhide. Virtualenv now prefers "venv" scheme if exists and because we need the patch to not affect virtualenv there is a new "venv" scheme as a copy of the original one before we add the `/local` prefixes to it.

Manually tested and it seems to work fine. Builds are available in: https://copr.fedorainfracloud.org/coprs/lbalhar/venv_scheme/builds/

If approved, I'm gonna squash this commit to the original one and force-push it to the 3.10 and 3.11 branches.

### CODE REVIEW

File: Lib/sysconfig.py [hunk 1]
```diff
@@ -61,6 +61,10 @@
 # backup the original posix_prefix as rpm_prefix
 # RPM packages use it and we need to be able to read it even when changed
 _INSTALL_SCHEMES['rpm_prefix'] = _INSTALL_SCHEMES['posix_prefix']
+# Virtualenv from 20.10.0 favors a new "venv" scheme over the defaults.
+# See: https://github.com/pypa/virtualenv/commit/8da79db86d8a5c74d03667a40e64ff832076445e
+# "venv" should be the same as the default for us.
```
Comment (hroncok) [inline Lib/sysconfig.py]:
```suggestion
# "venv" should be the same as the unpatched posix_prefix for us,
# so new virtual environments aren't created with paths like venv/local/bin/python.
```

**Comment by frenzymadness** [comment]
This patch seems to work for me with Python 3.10 and 3.11 - tested installation of packages in virtualenv and outside of it - both methods work as expected.

File: Lib/sysconfig.py [hunk 2]
```diff
@@ -61,6 +61,10 @@
 # backup the original posix_prefix as rpm_prefix
 # RPM packages use it and we need to be able to read it even when changed
 _INSTALL_SCHEMES['rpm_prefix'] = _INSTALL_SCHEMES['posix_prefix']
+# Virtualenv from 20.10.0 favors a new "venv" scheme over the defaults.
+# See: https://github.com/pypa/virtualenv/commit/8da79db86d8a5c74d03667a40e64ff832076445e
```
Comment (hroncok) [inline Lib/sysconfig.py:65]:
```suggestion
# See: https://github.com/pypa/virtualenv/commit/8da79db86d8a5c74d03667a40e64ff832076445e
# See: https://bugs.python.org/issue45413
```

File: Lib/sysconfig.py [hunk 3]
```diff
@@ -61,6 +61,10 @@
 # backup the original posix_prefix as rpm_prefix
 # RPM packages use it and we need to be able to read it even when changed
 _INSTALL_SCHEMES['rpm_prefix'] = _INSTALL_SCHEMES['posix_prefix']
+# Virtualenv from 20.10.0 favors a new "venv" scheme over the defaults.
```
Comment (hroncok) [inline Lib/sysconfig.py]:
```suggestion
# Virtualenv >= 20.10.0 favors the "venv" scheme over the defaults when creating virtual environments.
```

**Review by hroncok — COMMENTED** [review]
The code change is well tested and it works. Added several suggestions to the comment. But I suggest somebody who is not familiar with the problem/solution reviewing the comment for understandability.

**Comment by frenzymadness** [comment]
@hrnciar could you please check whether the comment in sysconfig.py makes sense to you?

**Comment by hrnciar** [comment]
@frenzymadness +1, the comment makes sense to me.

**Comment by frenzymadness** [comment]
So, this update is now in the fedora-3.11 branch with overwritten tag fedora-3.11.0a2-1 so it can be included in https://src.fedoraproject.org/rpms/python3.11/pull-request/7 and also in fedora-3.10 with a new tag fedora-3.10.0-3 which I use to create a new PR with. @hroncok could you please check that I did everything correctly?

**Comment by frenzymadness** [comment]
Hmmm, importpatches respects the order of commits. I've changed their order so the latest updated is the latest one and also the newest tag points to the latest commit in the branch but now the importpatches changes the order of patches in the specfile. Is that a problem?

**Comment by hroncok** [comment]
> Is that a problem?

Well, I'd rather keep the order sorted by patch numbers. It's confusing when 328 is applied before 251.

**Comment by hroncok** [comment]
Note that 251 was already changed more recently than 328 before this change and the order was not changed. Changing it now redefines the workflow -- I'd happily discuss changes in the workflow, but I am not happy to mix them with actual patch changes.

**Comment by frenzymadness** [comment]
All done in fedora-3.1{0,1} branches.

**Pull request closed without merging**